### ISSUE TITLE
TupleGCSStoreBackend empties the bucket it is configured with

### ISSUE THREAD

**Issue by JonathanCiglia** [fivetran/great_expectations#2384]
Hello !

**Describe the bug**

I have setup _great_expectations_ within a JupyterLab and on my local machine, with the same configuration.
I have declared two new data_docs_sites:

```yaml
...
  local_site:
    class_name: SiteBuilder
    store_backend:
      class_name: TupleFilesystemStoreBackend
      base_directory: uncommitted/data_docs/local_site/
    site_index_builder:
      class_name: DefaultSiteIndexBuilder
  gs_site:
    class_name: SiteBuilder
    show_how_to_buttons: false
    store_backend:
      class_name: TupleGCSStoreBackend
      project: my_project_name
      bucket: my_bucket_name
```

I have created an expectation suite and shared the .json file among my 2 contexts (JupyterLab & on my local machine).

I have two actions `StoreValidationResultAction` & `UpdateDataDocsAction` that I trigger whenever I want to run an expectation suite against a given batch.

It works well BUT, when I switch from my local machine to the JupyterLab (or vice-versa of course) and trigger these actions, the whole bucket I have setup is being emptied ! Only an `index.html` is created.

If I re-run this process (from the same platform), then the different folders (`validations/`, `expections/` ...) are created and everything is fine (although my whole bucket has been emptied).

**Expected behavior**

I would expect great_expectations to fill the existing folders.
I would also expect great_expectations to prevent emptying a whole bucket, or at least, have a validation from the user.
Maybe force the use of `prefix` to prevent deletion of unrelated great_expectations stuff (solution not tested)

**Environment (please complete the following information):**
 - Operating System: JupytyerLab 2.2.4 & debian for my local instance
 - Great Expectations Version: 0.13.8

**Comment by eugmandel** [comment]
Thank you for submitting this, @JonathanCiglia - we will triage this issue tomorrow.

**Comment by eugmandel** [comment]
@JonathanCiglia We triaged this issue today - the core team will pick it up for investigation.

**Comment by anthonyburdi** [comment]
@JonathanCiglia Thank you so much for bringing this issue to our attention. I had closed an earlier very similar issue #2121 as I was not able to replicate it. However I was able to in this instance. The way I replicated was to delete the validations in my validations store. Then when running a validation operator with the `UpdateDataDocsAction` there are no validations to use besides the one that was just created and so the `UpdateDataDocsAction` removes all of the other rendered html validations. Great Expectations creates data docs using the validations store contents.

What I think may be happening in your case is that your validation stores are pointing to different locations in your different environments. So when you switch to another environment, it is similar to what happens when I deleted my validations store - your validations store in the environment you switched to is empty. 

The best way to remedy this is to use the same validation store for both environments. You may wish to do that in a common location like a GCS bucket, or any one of the other options for storing metadata we support: https://docs.greatexpectations.io/en/latest/guides/how_to_guides/configuring_metadata_stores.html (You can also duplicate validations in two stores using your sync tool of choice if you prefer to have separate validation stores)

I'm going to close this issue but of course if I am wrong in my assumption about your validation store configuration or you try this and are still having issues, please let me know!
